src/kafka-util/src/client.rs

```diff
@@ -239,18 +244,36 @@ pub struct BrokerRewrite {
 #[derive(Clone)]
 pub struct BrokerRewritingClientContext<C> {
     inner: C,
-    rewrites: BTreeMap<BrokerAddr, Arc<dyn Fn() -> BrokerRewrite + Send + Sync>>,
+    rewrites: Arc<Mutex<BTreeMap<BrokerAddr, Arc<dyn Fn() -> BrokerRewrite + Send + Sync>>>>,
+    default_override: Option<(GlobalId, SshTunnelConfig)>,
```

Comment:
This API was designed to be oblivious to SSH tunneling for separation of responsibilities. I think it should be possible to follow the existing pattern here of using a closure:

```suggestion
    default_rewrite: Option<Arc<dyn Fn() -> BrokerRewrite + Send + Sync>>
```

Also `default_rewrite` rather than `default_override` to match with the existing "rewrite" terminology.

And then the SSH tunnel manager stuff moves into the connection.rs code, in a closure when you call `add_default_broker_rewrite`.

Reply:
This context is going to have to have a deeper understanding of ssh tunnels (so it can report errors correctly) in the next pr, once i adjust that to the new, simpler model @petrosagg and I came up with, so i will probably leave this!